### PR TITLE
Implementation for disallowing tuple unpacking of strings

### PR DESCRIPTION
**Pull request by akerami**
Implementation of #6406. 

Tests are failing, but /i believe the fix isn't too far from the implementation I have so far.

I would really appreciate suggestions on a different way of checking the type of a string so that errors do not occur with the tests.

Thanks!

### CODE REVIEW

File: mypy/checker.py [hunk 1]
```diff
@@ -2553,6 +2553,8 @@ def check_multi_assignment(self, lvalues: List[Lvalue],
         elif isinstance(rvalue_type, UnionType):
             self.check_multi_assignment_from_union(lvalues, rvalue, rvalue_type, context,
                                                    infer_lvalue_type)
+        elif rvalue_type.type.fullname == 'builtins.str':
```
Comment (JukkaL) [inline mypy/checker.py]: You need to protect this with an `isinstance` check.

Comment (akerami) [reply mypy/checker.py]: Yeah it was my original thought to match on `isinstance(rvalue_type, str)` but then this conditional branch will not be triggered even though `rvalue_type.type.fullname == 'builtins.str' ` does get triggered.

Comment (msullivan) [reply mypy/checker.py]: Protected with an `isinstance(rvalue_type, Instance)` check, probably. Note that checking mypy with itself produces a type error now.

Comment (akerami) [reply mypy/checker.py]: I see thanks, I made the change, it made most of the tests pass for me, except a few cases where the actual output of a test included the error message which made the tests fail due to a mismatch.

**Comment by akerami** [comment]
So now that the checks have passed should I write unit tests in test-data?

**Comment by msullivan** [comment]
Yes please.

**Comment by akerami** [comment]
Not sure where to place my tests, should I place it right under `[case testUnionOfNonIterableUnpacking]` in `check-unions.test`?

**Comment by msullivan** [comment]
That sounds good, yeah.

**Comment by akerami** [comment]
Okay are there any further points of improvements that should be made?

**Comment by davidzwa** [comment]
@JukkaL we're eager to hear your thoughts.

**Pull request merged**